### PR TITLE
FAI-809: Prepare bindings for repo migration

### PR DESCRIPTION
**Pull request by ruivieira**
Prepare `trustyai-explainability-python` for the `trustyai-explainability` repository migration.
At the moment, GHA CI checks are performed using the PyPi release of TrustyAI which includes the last Maven Central-released explainability (part of Kogito).
The goal of this PR is to enable CI PR checks to be performed using the latest `trustyai-explainability:main`.

### Changes

- Namespaces changed from `org.kie.kogito.*` to `org.kie.trustyai.*`
- GHA perform CI agains the latest `trustyai-explainability:main` by:
  - Fetch latest `trustyai-explainability:main` and build the shaded JAR
  - Build shaded `arrow-converter`
  - Bundle and install locally TrustyAI bindings
  - Test/Lint/Format
  - Use Java exp-core test models to stop bundling the test JARs

```mermaid
graph TD
    subgraph trustyai-explainability
    commitA[abc123] --> commitB[efg456]
    commitB[efg456] --> commitC["hij789 (HEAD)"]
    end
    subgraph trustyai-explainability-python
    PR["New PR"] --> build["Build TrustyAI package"]
    subgraph build-trustyai
    commitC -->fetch["Build latest shaded JAR"]
    fetch --> bundle
    buildArrow["Build latest Arrow"] --> bundle
    bundle --> build
    end
    build --> test["Test/Lint/Format"]
    end
  ```

### CODE REVIEW

File: trustyai/initializer.py [hunk 1]
```diff
@@ -16,41 +16,18 @@
...
     except AttributeError:
         default_dep_path = os.path.join(get_python_lib(), "trustyai", "dep")
 
     core_deps = [
-        f"{default_dep_path}/org/kie/kogito/explainability-core/{trusty_version}/*",
-        f"{default_dep_path}/org/slf4j/slf4j-api/1.7.30/slf4j-api-1.7.30.jar",
-        f"{default_dep_path}/org/apache/commons/commons-lang3/3.12.0/commons-lang3-3.12.0.jar",
-        f"{default_dep_path}/org/optaplanner/optaplanner-core-impl/8.22.1.Final/"
-        f"optaplanner-core-impl-8.22.1.Final.jar",
-        f"{default_dep_path}/org/apache/commons/commons-math3/3.6.1/commons-math3-3.6.1.jar",
```
Comment (danielezonca) [inline trustyai/initializer.py:32]: Nice improvement :)

File: .github/actions/build-arrow-converter/action.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+name: Build arrow-converter JAR
+runs:
+  using: "composite"
+  steps:
+    - name: Build arrow-converter
+      shell: bash
+      run: |
+          mvn compile package -DskipTests -f java_sources/trusty-ai-arrow/pom.xml
+          mv java_sources/trusty-ai-arrow/target/arrow-converters-0.0.1.jar trustyai/dep/org/trustyai/
```
[the remark this answers is not in the document]
Comment (ruivieira) [reply .github/actions/build-arrow-converter/action.yml:9]: Done

File: tests/general/test_counterfactualexplainer.py [hunk 3]
```diff
@@ -125,7 +125,7 @@ def test_counterfactual_v2():
     model = Model(predict_function, dataframe=True)
     goal = np.array([[0]])
     prediction = counterfactual_prediction(input_features=features, outputs=goal)
-    explainer = CounterfactualExplainer(steps=10_000)
+    explainer = CounterfactualExplainer(steps=100_000)
```
[the remark this answers is not in the document]
Comment (ruivieira) [reply tests/general/test_counterfactualexplainer.py:128]: The test is flaky, but only with the latest `expl-core`.

File: java_sources/trusty-ai-arrow/pom.xml [hunk 4]
```diff
@@ -3,53 +3,79 @@
...
-            <groupId>com.google.flatbuffers</groupId>
-            <artifactId>flatbuffers-java</artifactId>
-            <version>2.0.3</version>
-        </dependency>
-    </dependencies>
+  <dependencies>
+    <dependency>
+      <groupId>org.kie.trustyai</groupId>
+      <artifactId>explainability-core</artifactId>
+      <version>2.0.0-SNAPSHOT</version>
```
[the remark this answers is not in the document]
Comment (ruivieira) [reply java_sources/trusty-ai-arrow/pom.xml:22]: https://issues.redhat.com/browse/FAI-832

**Review by tteofili — APPROVED** [review]
huge PR, LGTM, thanks @ruivieira !

**Pull request merged**